File: MyMoney.Core/Models/AccountDashboardDisplayItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using CommunityToolkit.Mvvm.ComponentModel;

namespace MyMoney.Core.Models
{
    public partial class AccountDashboardDisplayItem : ObservableObject
    {
        [ObservableProperty]
        private string _accountName = "";

        [ObservableProperty]
        private Currency _total = new Currency(0m);

        public static AccountDashboardDisplayItem FromAccount(Account account)
        {
            var displayItem = new AccountDashboardDisplayItem();
            displayItem.AccountName = account.AccountName;
            displayItem.Total = account.Total;
            return displayItem;
        }

        public static AccountDashboardDisplayItem FromInitializers(string accountName, Currency total)
        {
            var displayItem = new AccountDashboardDisplayItem();
            displayItem.AccountName = accountName;
            displayItem.Total = total;
            return displayItem;
        }
    }
}

File: MyMoney.Core.FS/Models/AccountDashboardDisplayItem.fs
namespace MyMoney.Core.FS.Models

type AccountDashboardDisplayItem() =
    let mutable _AccountName = ""
    let mutable _Total = Currency(0m)

    member this.AccountName 
        with get () = _AccountName
        and set (value) = _AccountName <- value

    member this.Total
        with get () = _Total
        and set (value) = _Total <- value

    static member FromAccount(account: Account):AccountDashboardDisplayItem =
        let displayItem = AccountDashboardDisplayItem()
        displayItem.AccountName <- account.AccountName
        displayItem.Total <- account.Total

        displayItem

    member this.FromInitializers(accountName: string, total: Currency):AccountDashboardDisplayItem =
        let displayItem = AccountDashboardDisplayItem()
        displayItem.AccountName <- accountName
        displayItem.Total <- total

        displayItem
